src/api/uploads/BaseMultiput.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @flow
+ * @file Multiput upload base class
+ * @author Box
+ */
+import Base from '../Base';
+import type { MultiputConfig } from '../../flowTypes';
+
+const DEFAULT_MULTIPUT_CONFIG: MultiputConfig = {
+    console: false, // Whether to display informational messages to console
+    digestReadahead: 5, // How many parts past those currently uploading to precompute digest for
+    initialRetryDelayMs: 5000, // Base for exponential backoff on retries
+    maxRetryDelayMs: 60000, // Upper bound for time between retries
+    parallelism: 5, // Maximum number of parts to upload at a time
+    requestTimeoutMS: 120000, // Idle timeout on part upload, overall request timeout on other requests
+    // eslint-disable-next-line max-len
+    retries: 5 // How many times to retry requests such as upload part or commit. Note that total number of attempts will be retries + 1 in worst case where all attempts fail.
+};
+
+class BaseMultiput extends Base {
+    config: MultiputConfig;
+    sessionEndpoints: Object;
+
+    /**
+     * [constructor]
+     *
+     * @param {Object} options
+     * @param {Object} sessionEndpoints
+     * @param {MultiputConfig} [config]
+     * @return {void}
+     */
+    constructor(options: Object, sessionEndpoints: Object, config?: MultiputConfig): void {
+        super(options);
+
+        this.config = config || DEFAULT_MULTIPUT_CONFIG;
+        this.sessionEndpoints = sessionEndpoints;
+    }
+
+    /**
+     * If console logging is enabled in config, log a message to console
+     * 
+     * @private
+     * @param {string} msg
+     * @return {void}
+     */
+    consoleLog = (msg: string): void => {
+        if (this.config.console && window.console) {
+            // eslint-disable-next-line no-console
+            console.log(`${new Date().toString()} ${msg}`);
+        }
+    };
+
+    /**
+     * If console logging is enabled in config, call passed in function to generate a message and log it
+     * to console.
+     * 
+     * @private
+     * @param {function} msgFunc
+     * @return {void}
+     */
+    consoleLogFunc = (msgFunc: Function): void => {
```

Comment:
Whats the use case of this?

Reply:
@klk- do you remember why this was needed?

Reply:
It looks like we use this in event/error handlers where something needs to be dynamically calculated then for logging, see: https://github.com/box/box-ui-elements/pull/127/files#diff-ea9a71cb823147e0f8100d32ad629368R203

Reply:
But doesn't this function evaluates the msgFunc immediately and console.log the returned string? Why do we need to wrap it in another function?

Reply:
Right, maybe this is to be consistent with `consoleLog` above and wrap the logic for checking whether the config allows logging. I was thinking of just having one log method that takes either a string log or a log function:

```
logToConsole(msgOrMsgFunc: string | Function): void => {}
```

Reply:
Looking at all the use cases for `consoleLogFunc` I'm still not sure what do we get from it. 
```
this.consoleLogFunc(() => `Upload completed: ${this.stringify()}. Parts state: ${this.getPartsState()}`);
```

is the same as 
```
this.consoleLog(`Upload completed: ${this.stringify()}. Parts state: ${this.getPartsState()}`);
```
 right?

Reply:
Right...

Reply:
oh wait maybe it's because `getPartsState()` is expensive, and the point of having `consoleLogFunc` is to avoid calling that unnecessarily